src/include/gurt/debug.h

```diff
@@ -152,6 +152,14 @@ extern void (*d_alt_assert)(const int, const char*, const char*, const int);
 			D_DEBUG(flag_false, __VA_ARGS__);	\
 	} while (0)
 
+#define DL_CDEBUG(cond, flag_true, flag_false, _rc, _fmt, ...)                                     \
+	do {                                                                                       \
+		if (cond)                                                                          \
+			D_DEBUG(flag_true, _fmt ": " DF_RC " \n", ##__VA_ARGS__, DP_RC(_rc));      \
```

Comment:
Unrelated but seeing this made me think of VOS_TX_LOG_FAIL and daos_rpc_retryable_rc among others where we are checking many error codes to make a decision.  Perhaps it's not a big deal since they are only error paths but it seems like we could make it more efficient somehow, like via a bit mask or such.

Reply:
I just found out about the `VOS_TX_LOG_FAIL` macro today, I'll add it to the ones we check.

More eyes on this code, adding checkers and more consistency will make it easier to do more things with these macros.